tests/ert/ui_tests/gui/conftest.py

```diff
@@ -406,11 +406,6 @@ def wait_for_child(gui, qtbot: QtBot, typ: type[V], timeout=5000, **kwargs) -> V
     return get_child(gui, typ, **kwargs)
 
 
-def wait_for_attribute(gui_attribute, qtbot: QtBot, timeout=5000, **kwargs) -> V:
```

Comment:
The type hint should have been different and add an assert:
```python
 wait_for_attribute(gui_attribute: Callable[[], V | None], qtbot: QtBot, timeout=5000, **kwargs) -> V:
    qtbot.waitUntil(lambda: gui_attribute() is not None, timeout=timeout, **kwargs)
    t = gui_attribute()
    assert t is not None
    return t
```
in which case this method becomes quite a lot more useful as inline type hints will not complain about the possiblity of it being none.

Then you would call it like this:
```
 wait_for_attribute(lambda: run_dialog.fail_msg_box, qtbot)
```